laravel-live-coding/app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+
+class UserController extends Controller
+{
+    public function index()
+    {
+        $users = User::with('wallets.currency')->get()->map(function ($user){
+            return [
+                'name'=>$user->name,
+                'email'=>$user->email,
+                'phone'=>$user->phone,
+                'photo'=>$user->photo,
+                'wallets'=>$user->wallets->map(function ($wallet){
+                    return [
+                        'currency'=>$wallet->currency->name,
+                        'amount'=>$wallet->amount
+                    ];
+                })
+            ];
+        });
+        return response()->json($users);
+    }
+}
```

Comment:
It was a very bad solution...So we have UserController and method inside where you wrote realization.
But, what will you do, if you have to change the logic(for example add new fields)
You broke "O" principle you have to change your code
In this implementation, we don't have any flexibility and abstraction